test/functional/terminal/buffer_spec.lua

```diff
@@ -255,8 +256,11 @@ describe(':terminal buffer', function()
     command('bdelete!')
   end)
 
-  it('handles wqall', function()
+  it("won't close while a job is running (without !)", function()
     eq('Vim(wqall):E948: Job still running', exc_exec('wqall'))
+    command('call jobstop(&channel)')
+    retry(nil, nil, function() eq(-3, eval('jobwait([&channel], 0)[0]')) end)
+    command('bdelete')
```

Comment:
windows ci

```
[  ERROR   ] test/functional\terminal\buffer_spec.lua @ 259: :terminal buffer won't close while a job is running (without !)
  test\functional\helpers.lua:113: Vim(bdelete):E89: term://D:\a\neovim\neovim//1556:D:/a/neovim/neovim/build/bin/tty-test.EXE will be killed (add ! to override)

Reply:
I'm not sure why this is failing. If it makes it past the `retry` (meaning `jobwait()` returns `-3`) the job should be stopped...

Reply:
There's a race condition somewhere, as even running the test locally fails randomly.

Reply:
Using `feed_command` instead of `command` seems to fix it. My best guess is that `command` was being processed before the event loop called `process_close_event`.

Reply:
not sure we should accept that behavior. jobwait() should be reliable, otherwise plugin authors have to work around our bugs...

Reply:
I think I was wrong about the race condition. `f_jobwait` processes the job's event queue after the `process_wait` calls `decref` which enqueues `process_close_event`:

https://github.com/gpanders/neovim/blob/1ab8ead3a3efff0fca82e7cbdaef5648c235d28d/src/nvim/eval/funcs.c#L5414

But this leaves me back at square one. I'm no longer able to reproduce the failure on my dev machine, and it seems like it only fails on Windows, but I have no ideas why that would be.

Reply:
> And _even if_ the job still is valid, `f_jobwait` passes a different `MultiQueue` object to `process_wait`, so the `events` argument in the snippet above doesn't actually belong to `proc`

Nice catch, this looks like a mistake from f1de097dbb236ea400150f80b909407ca9af7441 .

Reply:
I put that particular change in a separate PR #15501 as I don't think that's our problem in this case. As I mentioned, `f_jobwait` isn't even always hitting `process_wait` since in a lot of cases the job is already stopped by the time we enter `f_jobwait`.

Checking the job directly [is implemented in this PR](https://github.com/neovim/neovim/pull/15402/files#diff-0b5b9677295695ef4fc7ce3c1a86a30c0c5633c5275ff1424a4e4219e01fa3fbR1241-R1245), though now the Windows CI is failing for a different reason :(

Reply:
> `f_jobwait` passes a different `MultiQueue` object to `process_wait`, so the `events` argument in the snippet above doesn't actually belong to `proc`.

Just to be clear (you probably know already): that "only" affects ordering, since `proc->events` is a child of `main_loop.events`.